src/detect-dsize.c

```diff
@@ -862,6 +869,96 @@ static int DsizeTestParse20 (void)
     return result;
 }
 
+/**
+ * \test this is a test for a valid dsize value !1
+ *
+ *  \retval 1 on succces
+ *  \retval 0 on failure
+ */
+static int DsizeTestParse21 (void)
+{
+    DetectDsizeData *dd = NULL;
+    dd = DetectDsizeParse("!1");
```

Comment:
See helper macros in `util-unittest.h` ... here, you'd use `FAIL_IF_NULL(dd)`.
Also, use `PASS` instead of return 1 or 0.